dev/integration_tests/deferred_components_test/run_release_test.sh

```diff
@@ -32,15 +32,22 @@ java -jar $bundletool_jar_path install-apks --apks=build/app/outputs/bundle/rele
 
 $adb_path shell "
 am start -n io.flutter.integration.deferred_components_test/.MainActivity
-sleep 30
+sleep 20
 exit
 "
-$adb_path logcat -d -t "$script_start_time" > build/app/outputs/bundle/release/run_logcat.log
-echo ""
-if cat build/app/outputs/bundle/release/run_logcat.log | grep -q "Running deferred code"; then
-  echo "All tests passed."
-  exit 0
-fi
-cat build/app/outputs/bundle/release/run_logcat.log
+x=20
+while [ $x -le 120 ]
+do
+  $adb_path logcat -d -t "$script_start_time" > build/app/outputs/bundle/release/run_logcat.log
+  if cat build/app/outputs/bundle/release/run_logcat.log | grep -q "Running deferred code"; then
+    echo "All tests passed."
+    exit 0
+  fi
+  echo "Waiting for $x seconds"
+  # We use an increasing delay here as the emulator can lag and take a long time to finish running the app.
+  # Delay of 20s produces ~5-10% flakes, 30s produces ~2% flakes roughly.
+  sleep 10
```

Comment:
I was trying some code, and this seems to do it:

```bash
#!/bin/bash

adb logcat | while read LOGLINE
do
    if [[ "${LOGLINE}" == *"Running deferred code"* ]]; then
        echo "Found ${LOGLINE}"
        pkill -P $$
    fi
done
```

/cc @gspencergoog who knows more about bash.

Reply:
It has to kill the logcat child process. I don't know if there's a more seamless way of doing it.

Reply:
Ahh thanks for this, this seems cleaner!

Reply:
Had to do some subshell wrangling, but this now pipes directly from adb!